### DISCUSSION TITLE
API call in getStaticProps/getServerSideProps

### DISCUSSION THREAD

**Discussion by kuohongg** [vercel/next.js#17668] [Help]
When I call API in getStaticProps/getServerSideProps, it returned error message to me stating the origin is not allowed. But when I call the API directly in my functional component, it return me the correct data. Any idea on this?

**Comment by francisrod01** [comment]
I haven't introduced any code as an example, so I guess it's because you should add your absolute URL every time you call for data by server-side. This new safe policy is needed to prevent Http host headers attack, for example.

See my comment here + code of how to prevent that:
https://stackoverflow.com/a/59631100/3332734
